### ISSUE TITLE
accidental symbol spill into libcsound64.so.6.0?

### ISSUE THREAD

**Issue by umlaeute** [csound/csound#1647]
When updating the Debian packages of Csound from 6.17 to 6.18, I noticed that a lot of new symbols appeared in `libcsound64.so.6.0`.

This wouldn't be an issue per se (of course you are free to add whatever symbols you like to this library), but there were a few indicators that hinted that this might not have been altogether intentional:
- the sheer number of added symbols
  approx. 2250 *new* symbols! in a "mainly bug fixing release"? In Csound-6.17, we had about 330 symbols in total!
- the symbol names themselves
  previously, practically all exported symbols had a `cs` prefix (with some variations, like `CS_`, `cs_`, `csound` and `cscore`). with the newly added symbols, about 300 have this prefix, leaving almost 2000 new symbols without a namespace ; some of these symbols are rather terse (e.g. `zir`).

I'm attaching the list as a ZIP-file (fiven it's size it's probably not a good idea to include it verbatim in this issue)
[newsymbols.zip](https://github.com/csound/csound/files/9902890/newsymbols.zip)

Anyhow, my question is: is this symbol inflation intentional? if so: will these symbols stay in future releases, and can I rely on [semantic versioning](https://semver.org) of the library in case the exported symbols will change in an incompatible way (e.g. whenever any of these symbols is removed)?

**Comment by umlaeute** [comment]
to be honest, now that i have looked at the changes between 6.17 and 6.18, I don't really understand where this symbol spillage comes from (there's no obvious candidate)

**Comment by vlazzarini** [comment]
- Maybe there's a rogue header somewhere? I really don't know, but I can look at my macos build to see if I can spot any of that. What's the command you used to list the number of symbols?

**Comment by vlazzarini** [comment]
I checked the develop branch (7.0) and the build does not export them. So it's something that happened specifically on the 6.x branch (csound6)

**Comment by vlazzarini** [comment]
Found it: line 505 of CMakeLists.txt should be

```
  set(CMAKE_C_FLAGS  "${CMAKE_C_FLAGS} -DSNDFILE_MP3")
```

instead of

```
  set(CMAKE_C_FLAGS  "-DSNDFILE_MP3")
```

Not sure if this requires a 6.18.1

**Comment by umlaeute** [comment]
thanks for finding that so quickly.
this also fixes a few other weirdnesses i was having.

> Not sure if this requires a 6.18.1

from my (Debian's) POV, i don't need a bugfix release for that (as i've just patched the 6.18.0 sources to include https://github.com/csound/csound/commit/1f0beb4eee41346c5f0ffa056c37ca2aad128ca3).
i cannot speak for the Csound ecosystem though.

**Comment by vlazzarini** [comment]
thanks, I'll keep that under review. Maybe a sources release.
